key.go

```diff
@@ -420,3 +420,29 @@ func GenerateRSAKeyWithExponent(bits int, exponent int) (PrivateKey, error) {
 	})
 	return p, nil
 }
+
+// DeriveKey generates a key and IV from given password and salt using openssl EVB_BytesToKey()
+func DeriveKey(cipher *Cipher, digest *Digest, salt []byte, password string, iterations int) (key, iv []byte, err error) {
+	key = make([]byte, cipher.KeySize())
+	iv = make([]byte, cipher.IVSize())
+
+	var saltPtr, ivPtr, passwordPtr, keyPtr *C.uchar
+	if len(salt) != 0 && salt != nil {
+		saltPtr = (*C.uchar)(unsafe.Pointer(&salt[0]))
+	}
+	if len(iv) != 0 && iv != nil {
+		ivPtr = (*C.uchar)(unsafe.Pointer(&iv[0]))
+	}
+	if iterations < 1 {
+		return nil, nil, errors.New("iterations count must be 1 or greater")
+	}
+	passwordSize := C.int(len([]byte(password)))
```

Comment:
len([]byte(password)) is the same as len(password) but the latter does not make an extra copy.